### PR TITLE
Do not apply SimplifyIfReturnBoolRector when there are comments in between the if statements

### PR DESCRIPTION
**Pull request by gnutix**
See https://github.com/rectorphp/rector/issues/1846#issuecomment-536216506.

### CODE REVIEW

**Comment by TomasVotruba** [comment]
Could you fix it at one go?

Collect comments of precious nodes and add it via $node->setComment()

**Comment by gnutix** [comment]
I don't think we should apply the Rector if there's a comment in there. Because, as in my fixture, the comment might apply to the last return statement, and not the last two ones. Which could mess things up. So I would go for checking in there's a comment in between the ifs that the Rector is trying to simplify, and if yes, abort the operation.

**Comment by TomasVotruba** [comment]
Agreed. Please send PR

**Comment by gnutix** [comment]
Ready for review. :)

File: packages/CodeQuality/tests/Rector/If_/SimplifyIfReturnBoolRector/Fixture/should_keep_comments.php.inc [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    public function both($var): bool
+    {
+        // Imagine this is a very important comment here that should not be removed.
+        // Imagine this is another very important comment here that should not be removed.
+        return is_string($var);
+    }
+}
+?>
```
Comment (TomasVotruba) [inline packages/CodeQuality/tests/Rector/If_/SimplifyIfReturnBoolRector/Fixture/should_keep_comments.php.inc]: If the code should remain the same, the part under `-----` can be fully removed here

File: packages/CodeQuality/tests/Rector/If_/SimplifyIfReturnBoolRector/Fixture/should_keep_doc_blocks.php.inc [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+        /**
+         * @todo Imagine this is another very important comment here that should not be removed.
+         */
+        return true;
+    }
+}
+?>
+-----
+<?php
```
Comment (TomasVotruba) [inline packages/CodeQuality/tests/Rector/If_/SimplifyIfReturnBoolRector/Fixture/should_keep_doc_blocks.php.inc]: Same here

File: packages/CodeQuality/tests/Rector/If_/SimplifyIfReturnBoolRector/Fixture/should_keep_comments.php.inc (hunk 1 above)
Comment (gnutix) [reply packages/CodeQuality/tests/Rector/If_/SimplifyIfReturnBoolRector/Fixture/should_keep_comments.php.inc]: It's not the same : the Rector changed the if conditions. Only the comments are the same, but I can't use a "one part" fixture file then.

File: packages/CodeQuality/tests/Rector/If_/SimplifyIfReturnBoolRector/Fixture/should_keep_doc_blocks.php.inc (hunk 2 above)
Comment (gnutix) [reply packages/CodeQuality/tests/Rector/If_/SimplifyIfReturnBoolRector/Fixture/should_keep_doc_blocks.php.inc]: Same answer ;)

File: packages/CodeQuality/tests/Rector/If_/SimplifyIfReturnBoolRector/Fixture/should_keep_comments.php.inc (hunk 1 above)
Comment (TomasVotruba) [reply packages/CodeQuality/tests/Rector/If_/SimplifyIfReturnBoolRector/Fixture/should_keep_comments.php.inc]: Ah, I see.

**Pull request merged**